src/som/vmobjects/SArray.java

```diff
@@ -113,6 +115,79 @@ public Object getStoragePlain() {
       return this.getObjectStorage(ValueProfile.createClassProfile());
     }
   }
+  
+  public void convertAllToObject(){
```

Comment:
This looks problematic.
I would try to avoid code duplication with SArray and for instance AtPutPrim.
This is also something that likely needs to be profiled and specialized.